[AArch64] Remove Automatic Enablement of FEAT_F32MM

Stylie777 opened the pull request:
When `+sve` is passed in the command line, if the Architecture being targeted is V8.6A/V9.1A or later, `+f32mm` is also added. This enables FEAT_32MM, however at the time of writing no CPU's support this. This leads to the FEAT_32MM instructions being compiled for CPU's that do not support them.

This commit removes the automatic enablement, however the option is still able to be used by passing `+f32mm`.

jthackray commented on llvm/docs/ReleaseNotes.rst:
```diff
@@ -74,6 +74,7 @@ Changes to the AMDGPU Backend
 
 Changes to the ARM Backend
 --------------------------
+* FEAT_F32MM is no longer activated by default when using `+sve` on v8.6-A or greater. The feature is still availble and can be using by adding `+f32mm` to the command line options.
```
Nit: typo, s/availble/available/

Stylie777 replied:
Updated

jthackray commented:
> When `+sve` is passed in the command line, if the Architecture being targeted is V8.6A/V9.1A or later, `+f32mm` is also added. This enables FEAT_32MM, however at the time of writing no CPU's support this. This leads to the FEAT_32MM instructions being compiled for CPU's that do not support them.
> 
> This commit removes the automatic enablement, however the option is still able to be used by passing `+f32mm`.

Tiny minuscule nit: should be "CPUs" not "CPU's".

Stylie777 committed:
[AArch64] Remove Automatic Enablement of FEAT_F32MM

When `+sve` is passed in the command line, if the Architecture being targeted is V8.6A/V9.1A or later, `+f32mm` is also added. This enables FEAT_32MM, however at the time of writing no CPUs support this. This leads to the FEAT_32MM instructions being compiled for CPUs that do not support them.

This commit removes the automatic enablement, however the option is still able to be used by passing `+f32mm`.

Stylie777 commented:
@jthackray I have updated the commit message as part of the most recent patch set.

jthackray reviewed (approved):
Approved, but you should probably find additional reviewer.

ostannard reviewed (approved):
I was the one who originally decided that f32mm should be automatically enabled like this, and in hindsight I agree that that was the wrong decision. GCC never implemented this, so this is making us more compatible, not less.

rgwott commented on llvm/unittests/TargetParser/TargetParserTest.cpp:
```diff
@@ -2487,10 +2480,10 @@ AArch64ExtensionDependenciesBaseCPUTestParams
-          "fp-armv8", "fullfp16", "fp16fml", "i8mm",    "lse",          "mte",
-          "pauth",    "perfmon",  "predres", "ras",     "rcpc",         "rdm",
-          "sb",       "neon",     "ssbs",    "sve",     "sve2-bitperm", "sve2"},
+         {"v9.2a",    "bf16",    "crc",  "dotprod",      "flagm", "fp-armv8",
```
Are these entries not better off one per line? The patch is modifying all lines anyway, might as well make it so that any future modifications would be more modular.

Stylie777 replied:
I would probably agree with you but the formatting here was assigned by `clang-format`. I also thing this is outside of the scope of this PR and should be done separately as this is a formatting change rather than a functionality change which this PR is focused on.
